contracts/plugins/assets/erc20/RewardableERC20.sol

```diff
@@ -36,39 +43,59 @@ abstract contract RewardableERC20 is IRewardable, ERC20 {
         _claimAccountRewards(msg.sender);
     }
 
+    function sync() external {
+        _claimAndSyncRewards();
+    }
+
     function _syncAccount(address account) internal {
         if (account == address(0)) return;
-        uint256 shares = balanceOf(account);
         uint256 accountRewardsPerShare = lastRewardsPerShare[account];
-        if (rewardsPerShare == accountRewardsPerShare) return;
-        uint256 delta = rewardsPerShare - accountRewardsPerShare;
-
-        // {qRewards} = {qRewards/share} * {qShare} / {qShare/share}
-        uint256 newRewards = (delta * shares) / one;
-        lastRewardsPerShare[account] = rewardsPerShare;
-        accumulatedRewards[account] += newRewards;
+        uint256 shares = balanceOf(account);
+        uint256 _accumuatedRewards = accumulatedRewards[account];
+        uint256 _rewardsPerShare = rewardsPerShare;
+        if (accountRewardsPerShare < _rewardsPerShare) {
+            uint256 delta = _rewardsPerShare - accountRewardsPerShare;
+            delta = (delta * shares) / one;
+            // {qRewards} = {qRewards/share} * {qShare} / {qShare/share}
+            _accumuatedRewards = _accumuatedRewards + delta;
```

Comment:
to avoid changing the units of `delta`, lets just combine these 2 lines into
```
// {qRewards} = {qRewards} + {qRewards/share} * {qShare} / {qShare/share}
_accumulatedRewards += (delta * shares) / one;
```